.github/workflows/dotnet.yml

```
@@ -14,10 +14,14 @@ jobs:
     permissions:
       contents: read
       pull-requests: write
+      id-token: write
+
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)